src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -49,8 +50,27 @@ def create_configstore(client,
                                             sku=Sku(name=sku),
                                             tags=tags,
                                             public_network_access=public_network_access,
-                                            disable_local_auth=disable_local_auth)
+                                            disable_local_auth=disable_local_auth,
+                                            soft_delete_retention_in_days=retention_days,
+                                            enable_purge_protection=enable_purge_protection,
+                                            create_mode=CreateMode.DEFAULT)
+
+    return client.begin_create(resource_group_name, name, configstore_params)
+
 
+def recover_deleted_configstore(cmd, client, name, resource_group_name=None, location=None, yes=False):
+    if resource_group_name is None or location is None:
+        metadata_resource_group, metadata_location = resolve_deleted_store_metadata(cmd, name)
```

Comment:
> metadata_resource_group, metadata_location = resolve_deleted_store_metadata(cmd, name) [](http://example.com/codeflow?start=7&length=87)

We need both location and RG for recovering a deleted store. There are three cases here:
1. User provides both location & RG: We use what they provided
2. User does not provide loc or rg: We find the loc and rg by calling list_deleted. Error out if multiple stores with same name.
3. User provides either RG or Location: We try to find the loc and rg (both) and error out if there are multiple matches. 
Can we be more intelligent in the third scenario? Instead of erroring out, can we match the user provided location or rg to figure out which store they wanted?

Reply:
Sure, we can be a little intelligent here.
I added the logic to match location or resource_group when provided so that we don't error out unnecessarily.